From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Juan Sanchez Fuentes <juanesf91@gmail.com>
Date: Fri, 29 Aug 2026 00:00:00 -0400
Subject: [PATCH] arm64: dts: sun55i-a527-cubie-a5e: enable SPI NOR

The Cubie A5E has a Winbond W25Q128 16MiB SPI NOR on SPI0 (PC2/PC3/PC4/PC12).
Enable it in the kernel DTS so /dev/mtd0 appears. Uses the existing
sun55i-a523.dtsi pinctrl (spi0_pc_pins + spi0_cs0_pc_pin) at 40MHz.

Signed-off-by: Juan Sanchez Fuentes <juanesf91@gmail.com>
Assisted-by: opencode:muse-spark-1.2-contributor-free
---
 arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts b/arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts
+++ b/arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts
@@ -344,6 +344,18 @@
 	vcc-pm-supply = <&reg_aldo3>;
 };
 
+&spi0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi0_pc_pins>, <&spi0_cs0_pc_pin>;
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+	};
+};
+
 &uart0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_pb_pins>;
-- 
Armbian
